docs/administrators/troubleshooting.rst

```diff
@@ -0,0 +1,251 @@
+   :width: 100%
+   :align: center
+
+
+.. _check_which_fact_didnt_resolve:
+
+Check which facts are not yet resolved
+--------------------------------------
+
+To find out which fact of a certain resource is not yet resolved, click on the magnifier in front of the resource in the
+``undefined`` state, as shown in the figure below.
+
+.. figure:: ./images/troubleshooting/resources_in_the_undefined_state.png
+   :width: 100%
+   :align: center
+
+The list of attributes of that resource, will contain one attribute which is marked as undefined (See figure below). This is the
+attribute that wasn't resolved yet. Track the source of this attribute down within the configuration model to find out why this
+attribute is undefined.
+
+.. figure:: ./images/troubleshooting/undefined_attribute.png
+   :width: 100%
+   :align: center
+
+
+.. _agent_doesnt_come_up:
+
+Agent doesn't come up
+=====================
+
+This section explains how to troubleshoot the problem where an agent is in the down state while it should be up. In the figure
+shown below, the agent vm1 is down.
+
+.. figure:: ./images/troubleshooting/agent_in_down_state.png
+   :width: 100%
+   :align: center
+
+Agents can be started in two different ways, either automatically by the inmanta server (auto-started agents) or manually
+(manually-started) agents. More information about the configuration of both types of agent can be found on
+:ref:`this page<configure_agents>`. The Section :ref:`Auto-started agents<autostarted_agents>` describes how to troubleshoot
+this issue for agents started by the Inmanta server. The Section :ref:`Manually-started agents<manually_started_agents>`
+describes how to troubleshoot this issue for agents that were started manually.
+
+.. _autostarted_agents:
+
+Auto-started agents
+-------------------
+
+An auto-started agent is only started when that agent is present in the ``autostart_agent_map`` environment setting. Verify that
+requirement via the settings tab of the inmanta dashboard as shown in the figure below.
+
+.. figure:: ./images/troubleshooting/environment_settings_autostart_agent_map.png
+   :width: 100%
+   :align: center
+
+When the ``autostart_agent_map`` is configured correctly, but the agent is still not up, read the logs of the auto-started agent
+. These logs can be found at the following location: ``<config.log-dir>/agent-<environment-id>.[log|out|err]``. The
+environment ID is present in the URL of the dashboard. More information about the different log files can be found
+:ref:`here<administrators_doc_logging>`. When reading those log files, pay specific attention to error
+messages and warnings that could explain why the agent is marked as down. Also, ensure that the name of the agent under
+consideration is added as an endpoint to the agent process. The log file should contain the following message when a certain
+agent is added as an endpoint to the process:
+
+.. code-block:: text
+
+    inmanta.agent.agent Adding endpoint <agent-name>
+
```

Comment:
Agents not starting could also be due to:

1. wrong bind-address
2. wrong server-address
3. authentication problems (e.g. not set a jwt source with sign=true)
4. ssl problems (endless source of problems)
5. otherwise unable to connect (e.g. no loopback)

Reply:
What do you mean with "unable to connect (e.g. no loopback)"

Reply:
if you configure a machine, but don't add a loopback device, packets the machine sends to itself don't arrive (even when they are sent to the external ip of the device itself). It is just an example of general network mis configuration, it could be firewall problem as well. Just like auth/ssl, it is a class of problems outside of the inmanta config itself.

Reply:
but in my experience 1,2,4 are the common ones